ReactWindows/ReactNative.Shared/Views/View/ReactViewManager.cs

```diff
@@ -92,6 +99,192 @@ protected override BorderedCanvas CreateViewInstance(ThemedReactContext reactCon
+                { "types", types }
+            };
+        }
+
+        private string GetAccessToken(StorageFile file)
+        {
+            var futureAccessList = StorageApplicationPermissions.FutureAccessList;
+            var existingEntries = futureAccessList.Entries;
+            var capacity = futureAccessList.MaximumItemsAllowed;
+            var now = DateTime.Now;
+
+            // Use more aggresive timeout when number of items exceed 100 or half of the capacity
+            // Note: 100 is an arbitary limit in case that OS reduces the capacity (it is 1000 in RS1). 
+            var expiry = existingEntries.Count >= capacity / 2 || existingEntries.Count >= 100 ?
+                now.AddDays(-1) :
+                now.AddDays(-7);
+
+            foreach (var entry in existingEntries)
+            {
+                if (DateTime.TryParse(entry.Metadata, out DateTime time) && time < expiry)
+                {
+                    futureAccessList.Remove(entry.Token);
+                }
+            }
+
+            return futureAccessList.Add(file, now.ToString(System.Globalization.CultureInfo.InvariantCulture));
+        }
+
+        private async void OnDragEnter(object sender, DragEventArgs args)
+        {
+            var view = sender as BorderedCanvas;
+            var data = await GetDataTransferInfo(args.DataView);
+
+            view.GetReactContext()
+                .GetNativeModule<UIManagerModule>()
+                .EventDispatcher
+                .DispatchEvent(new DragDropEvent(view.GetTag(), "topDragEnter", data));
+        }
+
+        private async void OnDragOver(object sender, DragEventArgs args)
+        {
+            var view = sender as BorderedCanvas;
+
+            // TODO: Send this event to JS, get response (how?) and
+            // set the AcceptedOperation value. The JS handler in
+            // MessagePanel merely does this:
+            //
+            //      e.dataTransfer.effectAllowed = 'copy';
+            //      e.dataTransfer.dropEffect = 'copy';
+            //
+            // so a simple `allowDrop` flag may be sufficient.            
+            args.AcceptedOperation = DataPackageOperation.Copy;
+
+            var data = await GetDataTransferInfo(args.DataView);
+
+            view.GetReactContext()
+                .GetNativeModule<UIManagerModule>()
+                .EventDispatcher
+                .DispatchEvent(new DragDropEvent(view.GetTag(), "topDragOver", data));
+        }
+
+        private async void OnDrop(object sender, DragEventArgs args)
+        {
+            var view = sender as BorderedCanvas;
+            var data = await GetDataTransferInfo(args.DataView, true);
+
+            view.GetReactContext()
+                .GetNativeModule<UIManagerModule>()
+                .EventDispatcher
+                .DispatchEvent(new DragDropEvent(view.GetTag(), "topDrop", data));
+        }
+
+        private async void OnDragLeave(object sender, DragEventArgs args)
```

Comment:
it looks like this and a few other methods doesn't use any instance fields, so they shouldn't be instance methods. my preference would be to extract them and DragDropEvent into their own file, but they should at least be static.

Reply:
Fixed